sources/web/datalab/polymer/components/auth-panel/auth-panel.ts

```diff
@@ -64,8 +67,11 @@ class AuthPanel extends Polymer.Element {
   _signInChanged(signedIn: boolean) {
     this._signedIn = signedIn;
     if (signedIn) {
-      this._userInfo = 'Signed in as ' + GapiManager.getSignedInEmail();
-      this._projectInfo = 'No project is set';  // TODO
+      GapiManager.getSignedInEmail()
+        .then((email: string) => {
+          this._userInfo = 'Signed in as ' + email;
+          this._projectInfo = 'No project is set';  // TODO
+        });
     }
     const ev = new Event('signInOutDone');
     this.dispatchEvent(ev);
```

Comment:
Should these two lines be inside the then, after line 73? That would make the Account menu close only after the email is retrieved. Might not make any difference, as I think the only path to _signInChanged is after auth is done, which would mean the email is always already loaded, which I think means the promise gets resolved before returning, so the code at line 71 will always run before the code at line 76 anyway. But I'm not positive about that.